src/Components/WebAssembly/WebAssembly/src/Infrastructure/JSInteropMethods.cs

```diff
@@ -35,7 +38,12 @@ public static Task DispatchEvent(WebEventDescriptor eventDescriptor, string even
         {
             var renderer = RendererRegistry.Find(eventDescriptor.BrowserRendererId);
             var jsonSerializerOptions = DefaultWebAssemblyJSRuntime.Instance.ReadJsonSerializerOptions();
-            var webEvent = WebEventData.Parse(renderer, jsonSerializerOptions, eventDescriptor, eventArgsJson);
+
+            // JsonSerializerOptions are tightly bound to the JsonContext. Cache it on first use using a copy
+            // of the serializer settings.
+            _jsonContext ??= new(new JsonSerializerOptions(jsonSerializerOptions));
```

Comment:
Why the double wrapped JsonSerializerOptions?

Reply:
It looks like if you call `new JsonContext(jsonSerializerOptions)` it's tied to that instance of JsonContext. We use these serializer options extensively for other serialization and I wanted to leave the default one "untouched".

Reply:
Might be good to add that as a comment.

Reply:
cc @layomia - any concerns here that the `JsonSerializerOptions` object gets mutated when passed to a `new JsonContext`?

Reply:
We have discussed allowing a one to many relationship between one options instance and multiple context instances, which would make using the `jsonSerializerOptions` itself possible. I'll bring it up in the next review so we can close on it.

However, it would still be optimal to make an options instance immutable (wrt serialization configuration, e.g. `PropertyNameCaseInsensitive`) once it has been bound with one or more context instances so that we can initialize the metadata reliably (knowing that the options won't change).